src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Connection/SqlConnectionInternal.cs

```diff
@@ -1746,66 +2042,76 @@ protected override void Activate(Transaction transaction)
             }
         }
 
-        // @TODO: Is this suppression still required
-        [SuppressMessage("Microsoft.Globalization", "CA1303:DoNotPassLiteralsAsLocalizedParameters")] // copied from Triaged.cs
-        protected override void ChangeDatabaseInternal(string database)
-        {
-            // MDAC 73598 - add brackets around database
-            database = SqlConnection.FixupDatabaseTransactionName(database); // @TODO: Should go to a utility method
-            Task executeTask = _parser.TdsExecuteSQLBatch(
-                $@"USE {database}",
-                ConnectionOptions.ConnectTimeout,
-                notificationRequest: null,
-                _parser._physicalStateObj,
-                sync: true);
+        protected override void CleanupTransactionOnCompletion(Transaction transaction) =>
+            DelegatedTransaction?.TransactionEnded(transaction);
```

Comment:
This movement is missing a comment, and it highlights a slight change in approach: rather than getting the value of a property into a local variable and null-coalescing on the local variable, we null-coalesce on the property directly. If the value of the property changes halfway through the operation, we'll be operating on two different values (or might get a NullReferenceException.)

This might be fine - SqlConnection isn't supposed to be used in a non-thread-safe manner, and this is the first way I can see which would trigger that behaviour - but I want to make sure it's being done intentionally.